src/js/components/stock-movement-wizard/PickPage.jsx

```diff
@@ -224,7 +224,7 @@ class PickPage extends Component {
             pickPageItem.picklistItems,
             item => suggestion['inventoryItem.id'] === item['inventoryItem.id'],
           );
-          if (_.isEmpty(pick) || (pick.quantityPicked !== suggestion.quantityPicked)) {
+          if (_.isEmpty(pick) || ((pick.quantityPicked !== suggestion.quantityPicked) && (pick['binLocation.id'] === suggestion['binLocation.id']))) {
```

Comment:
I think this still does not fix the problem, since value of the pick can be wrong because of this: item => suggestion['inventoryItem.id'] === item['inventoryItem.id']